### ISSUE TITLE
feat(optimize): add multi-dimensional optimization and gradient-based methods

### ISSUE THREAD

**Issue by copyleftdev** [copyleftdev/agent-calc#12]
## Summary

The `optimize` command currently supports 1D bounded minimization via Brent's method. Real data science and engineering optimization problems are almost always multi-dimensional.

## Current capability

```json
{ "intent": "minimize_1d", "objective": { "kind": "quadratic", "a": 1, "b": -6, "c": 9 }, "lower": -10, "upper": 10 }
// → minimizer: 3.0, minimum: 0.0 in 6 iterations (Brent's method)
```

### Gradient descent (unconstrained, n-dimensional)
```json
{
  "intent": "gradient_descent",
  "objective": "quadratic_form",
  "coefficients": [[2, 0], [0, 4]],
  "linear": [-4, -8],
  "constant": 0,
  "initial": [0, 0],
  "max_iterations": 1000,
  "tolerance": 1e-6
}
```

### Nelder-Mead simplex (derivative-free, n-dimensional)
```json
{
  "intent": "nelder_mead",
  "objective": { "kind": "rosenbrock" },
  "initial": [0, 0],
  "max_iterations": 5000
}
```

### Grid search (bounded, n-dimensional)
```json
{
  "intent": "grid_search",
  "objective": { "kind": "quadratic", "a": 1, "b": -6, "c": 9 },
  "bounds": [{ "lower": -10, "upper": 10 }],
  "resolution": 1000
}
```

### Constrained optimization (add to LP or new intent)
```json
{
  "intent": "minimize_constrained",
  "objective": [1, 1],
  "constraints": [
    { "coefficients": [1, 2], "relation": "ge", "rhs": 4 }
  ],
  "variables": [{ "lower": 0 }, { "lower": 0 }]
}
```

## Use cases

- **Data scientists**: hyperparameter optimization, loss surface exploration
- **Engineers**: multi-variable design optimization (weight × strength tradeoff)
- **Operations research**: non-linear constrained optimization beyond LP

## Notes

- `argmin` crate supports L-BFGS, SGD, Nelder-Mead — leverage existing dependency
- All results `approximate_f64` — flag consistently
- Add `iterations` and `converged` fields to all results for transparency
- Bounded-optimization invariant should extend: validate all bounds and initial points before entering solver
- Priority: `nelder_mead` (most general, already in argmin) → `gradient_descent` → `constrained`
